app/js/instances/instanceDetailController.js

```diff
@@ -21,32 +21,51 @@
 
 var _ = require('lodash');
 
-function instanceDetailController($scope, $routeParams, dockerService, instanceModel, instanceService) {
+function instanceDetailController($scope, $routeParams, $timeout, flux, dockerService, instanceModel, instanceService) {
     'use strict';
 
     // init
     $scope.containers = [];
     $scope.images = [];
+    $scope.loadingImages = {};
     $scope.instance = {alias: $routeParams.host};
-    $scope.allImages = false;
+    $scope.allImages = instanceModel.getShowAllImages();
     $scope.allContainers = false;
 
     dockerService.d('containers.list', {
-        host: $scope.instance.alias
+        host: $scope.instance.alias,
+        query: {
+            all: $scope.allContainers
+        }
     });
 
     dockerService.d('images.list', {
-        host: $scope.instance.alias
+        host: $scope.instance.alias,
+        query: {
+            all: $scope.allImages
+        }
     });
 
     // State event listeners
     $scope.$listenTo(instanceModel, function () {
         $scope.containers = instanceModel.getContainers();
         $scope.images = instanceModel.getImages();
+
+        var loadingImages = instanceModel.getLoadingImages();
+        _.forEach(loadingImages, function(progress, name) {
+            $scope.loadingImages[name] = progress;
+        });
+        $scope.loadingImages = _.pick($scope.loadingImages, _.keys(loadingImages));
+
+        $timeout(function() {
```

Comment:
Right now the reason it's wrapped in a timeout is because, ```$scope.$apply()``` angular will throw an exception if it runs directly in the ``` $scope.$listenTo```.